db/migrate/20170302134438_fix_only_superseded_editions.rb

```diff
@@ -1,16 +1,20 @@
 class FixOnlySupersededEditions < ActiveRecord::Migration[5.0]
   def up
-    Document
+    docs = Document
       .joins(:editions)
       .where(%q(
         NOT EXISTS (
           SELECT id FROM editions AS e
           WHERE e.state IN ('draft', 'published', 'unpublished')
             AND e.document_id = documents.id
         )
-      )).find_each do |document|
-        document.editions.order(updated_at: :desc).first
-          .update_column(:state, "unpublished")
-      end
+      ))
+
+    docs.find_each do |document|
+      document.editions.order(updated_at: :desc).first
+        .update_column(:state, "unpublished")
```

Comment:
If I recall correctly only a few of these have corresponding unpublishing objects and I believe we expect them to exist in other parts of the code, so would have thought this might go boom? you could try altering the unpublished edition factory to not include an unpublishing object and see what blows up in the tests?